pkg/azure/aks/machine_store.go

```diff
@@ -269,16 +265,31 @@ func (m *MachineStore) getMachineTypesByLocation(ctx context.Context, location s
 }
 
 func (m *MachineStore) CheckReadiness() bool {
-	return m.machineSizeMapLock.TryRLock() && m.machineMapLock.TryRLock()
+	// TODO - implement
+	return true
 }
 
-func (m *MachineStore) PopulateMachineStore(ctx context.Context) error {
+func (m *MachineStore) GetListOfVmsForSubscription() ([]*VirtualMachineInfo, error) {
+	m.machineMapLock.RLock()
+	defer m.machineMapLock.RUnlock()
+
+	vmi := make([]*VirtualMachineInfo, 0, len(m.MachineMap))
+	for _, vmInfo := range m.MachineMap {
+		vmi = append(vmi, vmInfo)
+	}
+
+	return vmi, nil
+}
+
+func (m *MachineStore) PopulateMachineStore() {
 	startTime := time.Now()
+	ctx := context.TODO()
```

Comment:
nit/suggestion: You should pass this context into the method.

Reply:
yup, good idea - done!